### ISSUE TITLE
Rule proposal: `consistent-variable-name`

### ISSUE THREAD

**Issue by erunion** [sindresorhus/eslint-plugin-unicorn#1587]
### Description

To enforce consistency, and prevent confusing alternatives, of specific variable names within a codebase. This would be much like [catch-error-name](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/catch-error-name.md) except that it would apply to general variable names.

### Fail

```js
var customer = 'me';
```

```js
var oas = require('./petstore.json');
```

### Pass

```js
const user = 'me';
```

```js
var apiDefinition = require('./petstore.json');
```

**Comment by sindresorhus** [comment]
In the meantime, you can use the `prevent-abbreviations` rule. Don't be mislead by the rule name. It can support any replacements.

This rule could probably reuse a lot of that machinery.

**Comment by fisker** [comment]
#1547 covers this?

**Comment by sindresorhus** [comment]
@fisker Only partly. It doesn't allow providing a replacement.

**Comment by erunion** [comment]
@sindresorhus I hadn't thought about functions, but agreed that it should apply to that too.
